mxcube3/routes/Beamline.py

```diff
@@ -84,25 +84,93 @@ def beamline_get_attribute(name):
     return Response(json.dumps(data), status=200, mimetype='application/json')
 
 
-@mxcube.route("/mxcube/api/v0.1/beamline/beamInfo", methods=['GET'])
+@mxcube.route("/mxcube/api/v0.1/beam/info", methods=['GET'])
 def getBeamInfo():
     """Beam information: position,size,shape
     return_data={"position":,"shape":,"size_x":,"size_y":}     
     """
+    ret = {}
+
+    beamInfo = mxcube.beamline.getObjectByRole("beam_info")
+    
+    if beamInfo is None:
+         logging.getLogger('HWR').error("beamInfo is not defined")
+         return Response(status=409)
+         
     try:
-        beamInfo = mxcube.beamline.getObjectByRole("beam_info")
-        if beamInfo is None:
-             logging.getLogger('HWR').error("beamInfo is not defined")
-             return Response(status=409)
         beamInfoDict = beamInfo.get_beam_info()
-        print beamInfoDict
-        data = {'position': beamInfo.get_beam_position(), \
-                'shape': beamInfoDict["shape"], \
-                'size_x': beamInfoDict["size_x"], \
-                'size_y': beamInfoDict["size_y"], \
-               }       
-        resp = jsonify(data)
+    except Exception:
+        beamInfoDict = dict()
+
+    try:
+        aperture = mxcube.diffractometer.getObjectByRole('aperture')
+        aperture_list = aperture.getPredefinedPositionsList()
+        current_aperture = aperture.getCurrentPositionName()
+    except Exception:
+        logging.getLogger('HWR').exception('could not get all Aperture hwobj')
+        aperture_list = []
+        current_aperture = None
+
+    try:
+        ret.update({'position': beamInfo.get_beam_position(),
+                    'shape': beamInfoDict.get("shape"),
+                    'size_x': beamInfoDict.get("size_x"),
+                    'size_y': beamInfoDict.get("size_y"),
+                    'apertureList' : aperture_list,
+                    'currentAperture' : current_aperture })
+    except Exception:
+        logging.getLogger("HWR").exception("Failed to get beam info")
+        return Response(status=409)
+    else:
+        resp = jsonify(ret)
         resp.status_code = 200
         return resp
+
+
+@mxcube.route("/mxcube/api/v0.1/diffractometer/movables/state", methods=['GET'])
```

Comment:
We talked here and agreed to also remove `movables` ; ok for moving the code to `Diffractometer.py`